quarkchain/cluster/grpc_client_test.py

```diff
@@ -0,0 +1,113 @@
+import unittest
+import grpc
+import grpc_testing
+from grpc.framework.foundation import logging_pool
+
+from quarkchain.cluster.grpc_client import GrpcClient
+from quarkchain.generated import grpc_client_pb2
+
+
+class TestClient(unittest.TestCase):
+    def setUp(self):
+        self.execution_thread = logging_pool.pool(1)
+
+        self.test_channel = grpc_testing.channel(
+            grpc_client_pb2.DESCRIPTOR.services_by_name.values(),
+            grpc_testing.strict_real_time(),
+        )
+
+    def tearDown(self):
+        self.execution_thread.shutdown(wait=False)
+
+    def test_set_root_chain_confirmed_block(self):
```

Comment:
I think the test cases in #813 are more readable and straightforward

Reply:
Still use the grpc testing but specify each corresponding situation in separate functions.